source/common/router/upstream_request.cc

```diff
@@ -427,6 +425,10 @@ void UpstreamRequest::onPoolReady(
         info.downstreamAddressProvider().connectionID().value());
   }
 
+  stream_info_.setUpstreamBytesMeterer(upstream_->bytesMeterer());
+  StreamInfo::StreamInfo::syncUpstreamAndDownstreamBytesMeterer(parent_.callbacks()->streamInfo(),
```

Comment:
I do not think this is correct, since (if I read the code correctly) it will cause the upstream bytes meter to be reset with each upstream request. I think the test that should pass is the following:

1. configure retries on specific error status code from upstream
2. return response with the error status code to cause Envoy to retry
3. return good response
4. validate that upstream byte count includes bytes for both 2. and 3.

Reply:
I think we can accumulate the previous measured bytes with the retried upstream request.